tasks/prime.py

```diff
@@ -615,24 +615,87 @@ def update_mto_shipment_status(self, overrides=None):
             logger.debug("updateMTOShipmentStatus: ⚠️ No mto_shipment found.")
             return None  # can't run this task
+        # To avoid issues with the mto shipment being stale
+        # retrieve the move associated with the shipment
+        # and then use the newly fetched move to the find most up to date version of the shipment
+        move_id = mto_shipment["moveTaskOrderID"]
+        headers = {"content-type": "application/json"}
+        resp = self.client.get(
+            support_path(f"/move-task-orders/{move_id}"),
+            name=support_path("/move-task-orders/{moveTaskOrderID}"),
+            headers=headers,
+            **self.cert_kwargs,
+        )
+        move_details, success = check_response(resp, "getMoveTaskOrder")
+        if not move_details:
+            logger.debug("updateMTOShipmentStatus: ⚠️ No mto_shipment found.")
+            return None  # can't run this task
+        for fetched_mto_shipment in move_details["mtoShipments"]:
+            if fetched_mto_shipment["id"] == mto_shipment["id"]:
 
+                # Generate fake payload based on the endpoint's required fields
+                payload = self.fake_request(
+                    "/mto-shipments/{mtoShipmentID}/status", "patch", SUPPORT_API_KEY, overrides
+                )
+
+                if fetched_mto_shipment["status"] == "CANCELLATION_REQUESTED" and payload["status"] != "CANCELED":
+                    return None
+                elif fetched_mto_shipment["status"] == "SUBMITTED" and payload["status"] not in [
+                    "APPROVED",
+                    "REJECTED",
+                ]:
+                    return None
+                elif fetched_mto_shipment["status"] == "DIVERSION_REQUESTED" and payload["status"] != "APPROVED":
+                    return None
+                elif fetched_mto_shipment["status"] == "APPROVED" and payload["status"] != "DIVERSION_REQUESTED":
+                    return None
+                elif fetched_mto_shipment["status"] in ["DRAFT", "REJECTED", "CANCELED"]:
+                    return None
+
+                headers = {"content-type": "application/json", "If-Match": fetched_mto_shipment["eTag"]}
+
+                resp = self.client.patch(
+                    support_path(f"/mto-shipments/{fetched_mto_shipment['id']}/status"),
+                    name=support_path("/mto-shipments/{mtoShipmentID}/status"),
+                    data=json.dumps(payload),
+                    headers=headers,
+                    **self.user.cert_kwargs,
+                )
+                new_mto_shipment, success = check_response(resp, "updateMTOShipmentStatus", payload)
+
+                if success:
+                    self.update_stored(MTO_SHIPMENT, mto_shipment, new_mto_shipment)
+                    return mto_shipment
+
+    @tag(MTO_SHIPMENT, "updateMTOShipmentStatus", "expectedFailure")
+    @task
+    def update_mto_shipment_with_invalid_status(self, overrides=None):
```

Comment:
It looks like this is getting run a lot more frequently than the real successful request (200 vs 40).  I wonder if we should try to limit the ratio of expected errors otherwise it may soon makeup a very large part of our requests.

Reply:
The number of times the successful request is run is artificially deflated since it first checks if the randomly generated shipment statuses are valid transitioning pairs and bails if they aren't. During the run that was captured in the screenshot, there was a range of 108-447

Reply:
It looks like we can provide a weight along with the @task decorator like `@task(1)`.  I don't know if it's valid to provide a non whole number like `.25` to maybe not run the expected failure task quite so much.

Reply:
<img width="838" alt="Screen Shot 2021-12-27 at 12 15 22 PM" src="https://user-images.githubusercontent.com/67110378/147493530-6b5e28f0-9829-4021-9b1c-63f7b303ad21.png">

I couldn't use a float but I was able to adjust the other support task to make the expected failure task run relatively less frequently